os/arch/arm/src/armv7-m/up_unblocktask.c

```diff
@@ -149,6 +152,11 @@ void up_unblock_task(struct tcb_s *tcb)
 			save_task_scheduling_status(rtcb);
 #endif
 
+			/* Restore the MPU registers in case we are switching to an application task */
+#ifdef CONFIG_TASK_SCHED_HISTORY
```

Comment:
Why do you split this from above even it has the same conditional?

Reply:
Fixed.